airflow/executors/inprocess_executor.py

```diff
@@ -0,0 +1,152 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""
+This module contains InProcessExecutor that is a single
+process executor meaning it does not use multiprocessing.
+"""
+
+import threading
+from typing import Any, Dict, List, Optional
+
+from airflow import conf
+from airflow.executors.base_executor import BaseExecutor
+from airflow.models.taskinstance import TaskInstance, TaskInstanceKeyType
+from airflow.utils.state import State
+
+
+class InProcessExecutor(BaseExecutor):
+    """
+    This executor is meant for debugging purposes. It is also can
+    be used with sqlite since sqlite.
+
+    It executes one task instance at time. Additionally to support working
+    with sensors, sensor's ``mode`` will be automatically set to "reschedule".
+    """
+
+    _terminated = threading.Event()
+
+    def __init__(self):
+        super().__init__()
+        self.tasks_to_run: List[TaskInstance] = []
+        # Place where we keep information for task instance raw run
+        self.tasks_params: Dict[TaskInstanceKeyType, Dict[str, Any]] = {}
+        self.fail_fast = conf.get("core", "fail_fast")
+
+    def execute_async(self, *args, **kwargs) -> None:
+        """
+        The method is replaced by custom trigger_task implementation.
+        """
```

Comment:
Should we raise exception or make an assert here?

Reply:
Why? This method is not used by the operator but must be implemented.